src/net_processing.cpp

```diff
@@ -5831,6 +5898,13 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
             std::vector<const CBlockIndex*> vToDownload;
             NodeId staller = -1;
             FindNextBlocksToDownload(*peer, MAX_BLOCKS_IN_TRANSIT_PER_PEER - state.vBlocksInFlight.size(), vToDownload, staller);
+            if (m_chainman.BackgroundSyncInProgress() && !IsLimitedPeer(*peer)) {
+                TryDownloadingHistoricalBlocks(
+                    *peer,
+                    MAX_BLOCKS_IN_TRANSIT_PER_PEER - state.vBlocksInFlight.size(),
```

Comment:
7b93d4adc976bdb32306951b21d7c98d9dacc972

Could use an assert this can't underflow

Reply:
Thanks, now avoiding underflow with cast + max.